src/AMSlib/wf/basedb.hpp

```diff
@@ -1700,9 +1822,74 @@ class RMQPublisherHandler : public AMQP::LibEventHandler
     std::future_status status;
     while ((status = future.wait_for(span)) == std::future_status::timeout &&
            (iters++ < repeat))
-      std::future<ConnectionStatus> established;
+      std::future<RMQConnectionStatus> established;
     return status == std::future_status::ready;
   }
+
+  /**
+   *  @brief  Free the data pointed pointer in a vector and update vector.
+   *  @param[in]  addr            Address of memory to free.
+   *  @param[in]  buffer          The vector containing memory buffers
+   */
+  void free_ams_message(int msg_id, std::vector<AMSMessage>& buf)
+  {
+    const std::lock_guard<std::mutex> lock(ptr_mutex);
+    auto it =
+        std::find_if(buf.begin(), buf.end(), [&msg_id](const AMSMessage& obj) {
+          return obj.id() == msg_id;
+        });
+    if (it == buf.end()) {
+      WARNING(RMQPublisherHandler,
+              "Failed to deallocate msg #%d: not found",
+              msg_id)
+      return;
+    }
+    auto& msg = *it;
+    auto& rm = ams::ResourceManager::getInstance();
+    try {
+      rm.deallocate(msg.data(), AMSResourceType::HOST);
+    } catch (const umpire::util::Exception& e) {
+      WARNING(RMQPublisherHandler,
```

Comment:
Do we want this to be a warning? or a fatal?

Reply:
Good question, I've put as a warning because you are "just" leaking memory but it's not fatal. if we want to be stricter I can move it as FATAL.

Reply:
My concern is that we may have bigger issues that we suppress here. E.g. We ask to de-allocate some region which was never allocated by Umpire. This can indicate some other error in the code.

Reply:
I think I have converged. I prefer being conservative here. And have it a fatal or a THROW.

Reply:
Got it. I have changed it for FATAL